### PR TITLE
images: use os-release for determining build-deps.sh parameters

### PR DESCRIPTION
**Pull request by jelly**
Use os-release to obtain the Fedora name / version so we can drop our Fedora-testing hack.

Tested locally by refreshing `fedora-testing`.

 * [x] image-refresh fedora-37

### CODE REVIEW

**Comment by martinpitt** [comment]
Thanks! Possible cleanup, and right now it's cheap to fix still.

File: images/scripts/fedora.setup [hunk 1]
```diff
@@ -163,14 +163,11 @@ if [ "${IMAGE%-testing}" != "$IMAGE" ]; then
     echo "config_opts['dnf.conf'] += '[updates-testing]\nenabled=1'" >>/etc/mock/default.cfg
 fi
 
-opsys=$(cut -d '-' -f 1 <<< "$IMAGE")
-version=$(cut -d '-' -f 2 <<< "$IMAGE")
-# If version is not number (ie: testing) then use Fedora 37
-if ! [ "$version" -eq "$version" ] 2>/dev/null; then version=37; fi
+. /etc/os-release
```
Comment (martinpitt) [inline images/scripts/fedora.setup:166]: I also changed this to /etc/, that's what the manpage recommends.

**Review by martinpitt — APPROVED** [review]
Thanks! I did the change above to eliminate the opsys/version variables. Doing a test refresh.

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-37-20230120-052121.log

**Comment by cockpituous** [comment]
image-refresh fedora-37 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-37-20230120-060217

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/fedora-37-20230120-060202.log

**Review by martinpitt — APPROVED** [review]
All green!

**Pull request merged**